apiserver/migrationtarget/migrationtarget.go

```diff
@@ -223,3 +224,70 @@ func (api *API) AdoptResources(args params.AdoptResourcesArgs) error {
 	}
 	return errors.Trace(env.AdoptResources(model.ControllerUUID(), args.SourceControllerVersion))
 }
+
+// CheckMachines compares the machines in state with the ones reported
+// by the provider and reports any discrepancies.
+func (api *API) CheckMachines(args params.ModelArgs) (params.ErrorResults, error) {
+	var empty params.ErrorResults
+	model, err := api.getModel(args.ModelTag)
+	if err != nil {
+		return empty, errors.Trace(err)
+	}
+	st, release, err := api.pool.Get(model.UUID())
+	if err != nil {
+		return empty, errors.Trace(err)
+	}
+	defer release()
```

Comment:
`State.GetModel()`  - used by `api.getModel()` - is about to go away (today?). It's also a super inefficient way to convert a tag to a model UUID which seems to be the only reason for calling it! 

Please use `names.ParseModelTag` and then get the UUID from that to pass to the pool.